lib/ecto/adapters/sql.ex

```diff
@@ -525,26 +523,26 @@ defmodule Ecto.Adapters.SQL do
 
   @doc false
   def reduce(repo, statement, params, mapper, opts, acc, fun) do
-    {pool, default_opts} = repo.__pool__
+    {repo_mod, pool, default_opts} = Ecto.Registry.lookup(repo)
```

Comment:
We should probably move `Ecto.Registry.lookup(repo)` to a private function (given you had to repeat it in many places in this PR).